core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -691,7 +691,10 @@ private static int makeRegexpFlags(String stringFlags) {
 
     /** SQL {@code RLIKE} function. */
     public boolean rlike(String s, String pattern) {
-      return cache.getUnchecked(new Key(0, pattern)).matcher(s).find();
+      // Since Spark 2.0, string literals (including regex patterns) are unescaped in SQL parser
```

Comment:
`rlike` also enabled in HIVE library, does the `rlike` function have the same semantics in hive as in spark sql?

Reply:
Yes, judging by the engine execution behavior, Hive also performs the unescape, and the result is consistent with Spark.
![image](https://github.com/apache/calcite/assets/7956306/0cf70e02-afd3-4b48-b0b1-5dcd6e86a204)